src/components/Comment/Comment.tsx

```diff
@@ -0,0 +1,155 @@
+      const replyData: NewComment = {
+        // will change after implementing authentication
+        author: {
+          name: "Emily Clark",
+          image:
+            "https://images.pexels.com/photos/774909/pexels-photo-774909.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
+        },
+        createdAt: Timestamp.now(),
+        text: data.comment,
+        likedBy: [],
+      };
+      addReply({ recipeId, commentId, replyData });
+    }
+    reset();
+    setOpenReply(false);
+  };
+
+  const toggleOpenReply = () => {
+    setOpenReply((prev) => !prev);
+  };
+
+  const handleLikeToggle = () => {
+    if (!recipeId || !commentId) return;
+    updateCommentLikes({
+      commentId,
+      recipeId,
+      userId,
+      currentLikedBy: likedBy,
+    });
+  };
+
+  return (
+    <>
+      <Divider />
+      <Box
+        sx={{
+          display: "flex",
+          gap: { xs: 3, sm: 4 },
+          alignItems: "center",
+          mt: 3,
+        }}
+      >
+        <Avatar
+          src={author.image}
+          alt={author.name}
+          sx={{
+            width: { xs: "3rem", sm: "4rem" },
+            height: { xs: "3rem", sm: "4rem" },
+          }}
+        />
+        <Box>
+          <Typography sx={{ fontSize: "14px", fontWeight: "bold" }}>
+            {author.name}
+          </Typography>
+          <Typography sx={{ fontSize: "14px", color: "secondary.light" }}>
+            {calculateTimeDifference(createdAt)}
+          </Typography>
+        </Box>
+      </Box>
+      <Box sx={{ my: 3, ml: { xs: 5, sm: 8, md: 12 } }}>
+        <Box sx={{ mb: 2 }}>
+          <Typography sx={{ mb: 2.5 }}>{text}</Typography>
+          {!reply && (
+            <>
+              <Button
+                onClick={toggleOpenReply}
+                startIcon={
+                  <InsertCommentOutlinedIcon sx={{ color: "primary.light" }} />
+                }
+                sx={{ textTransform: "none", color: "secondary.light" }}
+              >
+                Reply
+              </Button>
+
+              <Button
+                startIcon={
+                  likedBy.includes(userId) ? (
+                    <FavoriteIcon sx={{ color: "primary.light" }} />
+                  ) : (
+                    <FavoriteBorderOutlinedIcon
+                      sx={{ color: "primary.light" }}
+                    />
+                  )
+                }
+                disabled={isLiking}
+                onClick={handleLikeToggle}
+                sx={{ textTransform: "none", color: "secondary.light" }}
+              >
+                {likedBy.length}
+              </Button>
+            </>
+          )}
+        </Box>
+        {openReply && (
+          <Box
+            component="form"
+            onSubmit={handleSubmit(onSubmit)}
+            sx={{ display: "flex", flexDirection: "column", gap: 2 }}
+          >
+            <CommentTextfield name="comment" control={control} required />
+            <Button
```

Comment:
You are already handling the isSubmitting state well, but it might be worth extending the logic to disable the form while isSubmitting to prevent multiple submissions.
Additionally, disabling the reply button based on the isLiking state is a good user experience practice, which you’ve handled well.
<CommentTextfield name="comment" control={control} required disabled={isSubmitting} />
<Button disabled={isSubmitting} variant="contained" type="submit">
  {isSubmitting ? "Submitting..." : "Reply"}
</Button>